### PR TITLE
tinycbor: add doc.txt file

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

I noticed that the tinycbor package was missing a doc.txt, so here it is.

### Testing procedure

Read the documentation. Nagging on grammatical and spelling mistakes is fully allowed here!

### Issues/PRs references

depends on #9947

### CODE REVIEW

File: pkg/tinycbor/doc.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (danpetry) [inline pkg/tinycbor/doc.txt]: Have made a few changes as a PR: https://github.com/bergzand/RIOT/pull/2

Comment (bergzand) [reply pkg/tinycbor/doc.txt]: Merged!

File: pkg/tinycbor/doc.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @defgroup pkg_tinycbor TinyCBOR CBOR library
```
[the remark this answers is not in the document]
Comment (bergzand) [reply pkg/tinycbor/doc.txt]: Changed

File: pkg/tinycbor/doc.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * @ingroup  pkg
+ * @brief    Provides the TinyCBOR Concise Binary Object Representation (CBOR)
```
[the remark this answers is not in the document]
Comment (bergzand) [reply pkg/tinycbor/doc.txt]: And also changed

**Comment by bergzand** [comment]
@smlng addressed your issues, thanks for the review!

**Comment by bergzand** [comment]
> please rebase and squash

Rebased and squashed, thanks!

**Pull request merged**